### DISCUSSION TITLE
Dump of GB(A) Games not working

### DISCUSSION THREAD

**Discussion by Rougie85** [sanni/cartreader#862] [Need help]
Hi all,
I started assembling the cart reader few days back and now I am at the point where I can test it... and it isn't working ;)

Device starts up, everything is ok. 

But as soon as I am trying to dump a GB game, I get this error:

```
OSCR HW5 V12.8

[+] Game Boy
[+] Game Boy (Color)

HEADER CHECKSUM ERROR

Press Button to
ignore or clean
cart and try again

Title: SSCCEE
Revision: 0
Mapper: MBC1
ROM Size: 128 KB
Save Size: 8 KB

Press Button...
[+] Read ROM

Saving to GB/ROM/SSCCEE/7/...
[*******************]
Checksum: 8B9E != BEBE
Invalid Checksum
CRC32... 4B28147C -> Not found
```

Ok, the name of the game is "Soccer"... SSCCEE is pretty close ;)
I checked already for bridges, cleaned the cartridge (other GB games arent't working too).

Any further idea? I read already some other discussions about it... 

Thanks in advance, I really like the project and want to find out what is stored on the Satellaview 8M card ;)

**Comment by Ancyker** [comment]
Clean the cart pins. Test with other carts for the same system if possible. Test other systems as well. If only that cart doesn't work, it's likely just that cart. If the other systems work but that system does not, the below advice should be limited to only the slot for that system, otherwise, it applies to the slot as well as all the headers.

If the checksum is consistently bad there is probably a solder joint that is not connected, try to figure out which one and fix it. 

If it changes each time there's likely one or more bad solder joints somewhere, these will appear connected unless you know what a cold solder joint looks like. In that case, reflow all the solder joints and see if that helps.

**Comment by sanni** [comment]
Just a wild guess but since it repeats the same letter twice it's probably pin A0 that is not toggling correctly.

Run the self test, inspect the GB slot with a magnifier, sometimes there is a solder blob on the vertical part of the pins higher up above the solder pads, re-solder everything on the Cart Reader that is part of the A0 trace.

If you still don't find the issue open the failed dump in an hex editor and check if the same value repeating twice pattern is continuing through out the whole rom or if the pattern changes.

**Reply by Rougie85** [reply]
Thanks a lot, that was the issue. I checked Rayman (GBC) too, it was RRYYAA.
All other connections were fine, A0 is somehow broken on this PCB. I tried to resolder etc., nothing did work. So at the end I took a wire and connected the two pins (see screenshot "Broken").

![gb](https://github.com/sanni/cartreader/assets/46425745/8c609829-d1e2-4b51-94be-a1a922e4240e)

OSCR HW5 V12.8

[+] Game Boy

[+] Game Boy (Color)

Title: SOCCER
Revision: 0
Mapper: MBC1
ROM Size: 128 KB
Save Size: None

Press Button...

[+] Read ROM

Saving to GB/ROM/SOCCER/11/...
[*******************]
Checksum: BE05 -> OK
CRC32... C36B199B -> Soccer (Europe, Australia) (En,Fr,De) (SGB Enhanced).gb

**Comment by Rougie85** [comment]
I built the six slot adapter now.
Everything is working, expect the SNES slot. There I always get wrong checksum information, e.g. 0505 for Yoshis Island... which should be E10B. 
I checked the connections already with a multimeter, they look fine. Any idea which should be checked again?
Other slots (Famicom, N64, Genesis etc.) are working. So I guess it is the adapter itself?

**Reply by Rougie85** [reply]
Ok... found the issue... one of the pins was bent and had no contact

Now everything is working :)
